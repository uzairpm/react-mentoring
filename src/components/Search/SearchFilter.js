import React, {Component} from 'react';
import PropTypes from 'prop-types';

import Button from '../Common/Button';

const SearchFilter = (props) => (
    <div className="card">
        <div className="searchFilterBar">
            {props.resultCount} movies found
            <div className="float-right">
                <label className="mr-3">Sort by</label>
                <Button label="Release Date"
                    clickHandler={props.releaseDateClick}
                    className={"btn btn-sm mr-3 " + (props.releaseDate ? 'btn-dark' : 'btn-light')}/>
                <Button label="Rating"
                    clickHandler={props.ratingClick}
                    className={"btn btn-sm " + (props.releaseDate ? 'btn-light' : 'btn-dark')}/>
            </div>
        </div>
    </div>
);

SearchFilter.propTypes = {
    resultCount: PropTypes.number.isRequired,
    releaseDateClick: PropTypes.func.isRequired,
    ratingClick: PropTypes.func.isRequired,
    releaseDate: PropTypes.bool
};
export default SearchFilter;
