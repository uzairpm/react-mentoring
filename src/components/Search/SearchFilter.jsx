import React, {Component} from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Button from '../Common/Button';

const SearchFilterBar = styled.div`
    padding: 5px 15px;
`;


const SearchFilter = (props) => (
    <div className="card">
        <SearchFilterBar>
            {props.resultCount} movies found
            <div className="float-right">
                <label className="mr-3">Sort by</label>
                <Button label="Release Date"
                    clickHandler={props.releaseDateClick}
                    className={"btn btn-sm mr-3 " + (props.releaseDate ? 'btn-dark' : 'btn-light')}/>
                <Button label="Rating"
                    clickHandler={props.ratingClick}
                    className={"btn btn-sm " + (props.releaseDate ? 'btn-light' : 'btn-dark')}/>
            </div>
        </SearchFilterBar>
    </div>
);

SearchFilter.propTypes = {
    resultCount: PropTypes.number.isRequired,
    releaseDateClick: PropTypes.func.isRequired,
    ratingClick: PropTypes.func.isRequired,
    releaseDate: PropTypes.bool
};
export default SearchFilter;
