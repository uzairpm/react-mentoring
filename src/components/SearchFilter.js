import React, {Component} from 'react';

import Button from './Common/Button';

/* export default class SearchFilter extends Component {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <div className="card">
                <div className="searchFilterBar">
                    {this.props.resultCount} movies found
                    <div className="float-right">
                        <label className="mr-3">Sort by</label>
                        <Button label="Release Date"
                            clickHandler={this.props.releaseDateClick}
                            className={"btn btn-sm mr-3 " + (this.props.releaseDate ? 'btn-dark' : 'btn-light')}/>
                        <Button label="Rating"
                            clickHandler={this.props.ratingClick}
                            className={"btn btn-sm " + (this.props.releaseDate ? 'btn-light' : 'btn-dark')}/>
                    </div>
                </div>
            </div>
        );
    }
} */
const SearchFilter = (props) => {
    return (
        <div className="card">
            <div className="searchFilterBar">
                {props.resultCount} movies found
                <div className="float-right">
                    <label className="mr-3">Sort by</label>
                    <Button label="Release Date"
                        clickHandler={props.releaseDateClick}
                        className={"btn btn-sm mr-3 " + (props.releaseDate ? 'btn-dark' : 'btn-light')}/>
                    <Button label="Rating"
                        clickHandler={props.ratingClick}
                        className={"btn btn-sm " + (props.releaseDate ? 'btn-light' : 'btn-dark')}/>
                </div>
            </div>
        </div>
    );
};
export default SearchFilter;