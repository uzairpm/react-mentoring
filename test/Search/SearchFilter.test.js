import { shallow } from 'enzyme';
import React from 'react';

import SearchFilter from '../../src/components/Search/SearchFilter';

describe('Check Component creation', () => {
    const onRDClick = jest.fn();
    const onRClick = jest.fn();
    test('SearchFilter component snapshot test', () => {
        const element = shallow(<SearchFilter resultCount={3} releaseDate={true}
            releaseDateClick={onRDClick} ratingClick={onRClick} />);
        expect(element).toMatchSnapshot();
    });
    test('SearchFilter component snapshot test', () => {
        const element2 = shallow(<SearchFilter resultCount={2} releaseDate={false}
            releaseDateClick={onRDClick} ratingClick={onRClick} />);
        expect(element2).toMatchSnapshot();
    });
});
