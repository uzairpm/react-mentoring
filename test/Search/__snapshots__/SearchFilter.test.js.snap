// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter Component Snapshot test with default props sorted by Rating 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchFilter
    ratingClick={[MockFunction]}
    releaseDate={false}
    releaseDateClick={[MockFunction]}
    resultCount={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="searchFilterBar"
        style={
          Object {
            "padding": "5px 15px",
          }
        }
      >
        2
         movies found
        <div
          className="float-right"
        >
          <label
            className="mr-3"
          >
            Sort by
          </label>
          <Button
            className="btn btn-sm mr-3 btn-light"
            clickHandler={[MockFunction]}
            label="Release Date"
          />
          <Button
            className="btn btn-sm btn-dark"
            clickHandler={[MockFunction]}
            label="Rating"
          />
        </div>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          2,
          " movies found",
          <div
            className="float-right"
          >
            <label
              className="mr-3"
            >
              Sort by
            </label>
            <Button
              className="btn btn-sm mr-3 btn-light"
              clickHandler={[MockFunction]}
              label="Release Date"
            />
            <Button
              className="btn btn-sm btn-dark"
              clickHandler={[MockFunction]}
              label="Rating"
            />
          </div>,
        ],
        "className": "searchFilterBar",
        "style": Object {
          "padding": "5px 15px",
        },
      },
      "ref": null,
      "rendered": Array [
        2,
        " movies found",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="mr-3"
              >
                Sort by
              </label>,
              <Button
                className="btn btn-sm mr-3 btn-light"
                clickHandler={[MockFunction]}
                label="Release Date"
              />,
              <Button
                className="btn btn-sm btn-dark"
                clickHandler={[MockFunction]}
                label="Rating"
              />,
            ],
            "className": "float-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
                "className": "mr-3",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "btn btn-sm mr-3 btn-light",
                "clickHandler": [MockFunction],
                "label": "Release Date",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "btn btn-sm btn-dark",
                "clickHandler": [MockFunction],
                "label": "Rating",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="searchFilterBar"
          style={
            Object {
              "padding": "5px 15px",
            }
          }
        >
          2
           movies found
          <div
            className="float-right"
          >
            <label
              className="mr-3"
            >
              Sort by
            </label>
            <Button
              className="btn btn-sm mr-3 btn-light"
              clickHandler={[MockFunction]}
              label="Release Date"
            />
            <Button
              className="btn btn-sm btn-dark"
              clickHandler={[MockFunction]}
              label="Rating"
            />
          </div>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            2,
            " movies found",
            <div
              className="float-right"
            >
              <label
                className="mr-3"
              >
                Sort by
              </label>
              <Button
                className="btn btn-sm mr-3 btn-light"
                clickHandler={[MockFunction]}
                label="Release Date"
              />
              <Button
                className="btn btn-sm btn-dark"
                clickHandler={[MockFunction]}
                label="Rating"
              />
            </div>,
          ],
          "className": "searchFilterBar",
          "style": Object {
            "padding": "5px 15px",
          },
        },
        "ref": null,
        "rendered": Array [
          2,
          " movies found",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="mr-3"
                >
                  Sort by
                </label>,
                <Button
                  className="btn btn-sm mr-3 btn-light"
                  clickHandler={[MockFunction]}
                  label="Release Date"
                />,
                <Button
                  className="btn btn-sm btn-dark"
                  clickHandler={[MockFunction]}
                  label="Rating"
                />,
              ],
              "className": "float-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sort by",
                  "className": "mr-3",
                },
                "ref": null,
                "rendered": "Sort by",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "btn btn-sm mr-3 btn-light",
                  "clickHandler": [MockFunction],
                  "label": "Release Date",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "btn btn-sm btn-dark",
                  "clickHandler": [MockFunction],
                  "label": "Rating",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SearchFilter Component Snapshot test with default props sorted by Release date 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchFilter
    ratingClick={[MockFunction]}
    releaseDate={true}
    releaseDateClick={[MockFunction]}
    resultCount={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="searchFilterBar"
        style={
          Object {
            "padding": "5px 15px",
          }
        }
      >
        3
         movies found
        <div
          className="float-right"
        >
          <label
            className="mr-3"
          >
            Sort by
          </label>
          <Button
            className="btn btn-sm mr-3 btn-dark"
            clickHandler={[MockFunction]}
            label="Release Date"
          />
          <Button
            className="btn btn-sm btn-light"
            clickHandler={[MockFunction]}
            label="Rating"
          />
        </div>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          3,
          " movies found",
          <div
            className="float-right"
          >
            <label
              className="mr-3"
            >
              Sort by
            </label>
            <Button
              className="btn btn-sm mr-3 btn-dark"
              clickHandler={[MockFunction]}
              label="Release Date"
            />
            <Button
              className="btn btn-sm btn-light"
              clickHandler={[MockFunction]}
              label="Rating"
            />
          </div>,
        ],
        "className": "searchFilterBar",
        "style": Object {
          "padding": "5px 15px",
        },
      },
      "ref": null,
      "rendered": Array [
        3,
        " movies found",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="mr-3"
              >
                Sort by
              </label>,
              <Button
                className="btn btn-sm mr-3 btn-dark"
                clickHandler={[MockFunction]}
                label="Release Date"
              />,
              <Button
                className="btn btn-sm btn-light"
                clickHandler={[MockFunction]}
                label="Rating"
              />,
            ],
            "className": "float-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
                "className": "mr-3",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "btn btn-sm mr-3 btn-dark",
                "clickHandler": [MockFunction],
                "label": "Release Date",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "btn btn-sm btn-light",
                "clickHandler": [MockFunction],
                "label": "Rating",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="searchFilterBar"
          style={
            Object {
              "padding": "5px 15px",
            }
          }
        >
          3
           movies found
          <div
            className="float-right"
          >
            <label
              className="mr-3"
            >
              Sort by
            </label>
            <Button
              className="btn btn-sm mr-3 btn-dark"
              clickHandler={[MockFunction]}
              label="Release Date"
            />
            <Button
              className="btn btn-sm btn-light"
              clickHandler={[MockFunction]}
              label="Rating"
            />
          </div>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            3,
            " movies found",
            <div
              className="float-right"
            >
              <label
                className="mr-3"
              >
                Sort by
              </label>
              <Button
                className="btn btn-sm mr-3 btn-dark"
                clickHandler={[MockFunction]}
                label="Release Date"
              />
              <Button
                className="btn btn-sm btn-light"
                clickHandler={[MockFunction]}
                label="Rating"
              />
            </div>,
          ],
          "className": "searchFilterBar",
          "style": Object {
            "padding": "5px 15px",
          },
        },
        "ref": null,
        "rendered": Array [
          3,
          " movies found",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="mr-3"
                >
                  Sort by
                </label>,
                <Button
                  className="btn btn-sm mr-3 btn-dark"
                  clickHandler={[MockFunction]}
                  label="Release Date"
                />,
                <Button
                  className="btn btn-sm btn-light"
                  clickHandler={[MockFunction]}
                  label="Rating"
                />,
              ],
              "className": "float-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sort by",
                  "className": "mr-3",
                },
                "ref": null,
                "rendered": "Sort by",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "btn btn-sm mr-3 btn-dark",
                  "clickHandler": [MockFunction],
                  "label": "Release Date",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "btn btn-sm btn-light",
                  "clickHandler": [MockFunction],
                  "label": "Rating",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
